### DISCUSSION TITLE
Lost a secret key

### DISCUSSION THREAD

**Discussion by dev-nicelee** [outline/outline#1441] [General] [answered]
## Problem
The original SECRET_KEY in .env file is lost and cannot retrieve user authentication.

## Steps to reproduce

1. Change original SECRET_KEY 
2. Login to outline app

## Environment

- Self-hosted (AWS Lightsail Server)

**Comment by dev-nicelee** [comment, marked as answer]
This is resolved by resetting jwtSecret for all users.

I've made my own solution and it's in my github repository [reset-jwtsecret-outline](https://github.com/dev-nicelee/reset-jwtsecret-outline)

**Reply by tclemans** [reply]
@dev-nicelee could you post your solution here? Your "reset-jwtsecret-outline" repository link returns 404. 😞

**Comment by dmorawetz** [comment]
I solved this with the following steps:
1. Deleting the jwtSecret column for all users
2. Adding `await user.rotateJwtSecret()` right before https://github.com/outline/outline/blob/ba2bfc7c897e5a5f48553532bd81d2409b19c8dd/server/utils/authentication.ts#L59-L60. More specifically, I just edited `build/server/utils/authentication.js` directly in the docker container, so I do not have to compile anything.
3. Login with all users.
4. Revert the edit to authentication.js.

**Comment by fabienheureux** [comment]
This PR addresses this as well : https://github.com/outline/outline/pull/10238
